core/services/ocr2/plugins/ocr2keeper/evmregistry/v21/logprovider/provider.go

```diff
@@ -169,6 +188,26 @@ func (p *logEventProvider) Start(context.Context) error {
 			})
 		})
 
+		p.threadCtrl.Go(func(ctx context.Context) {
+			// sync filters with buffer periodically,
+			// to ensure that inactive upkeeps won't waste capacity.
+			ticker := time.NewTicker(bufferSyncInterval)
+			defer ticker.Stop()
+
+			for {
+				select {
+				case <-ticker.C:
+					if p.bufferV1 != nil {
```

Comment:
should this be conditioned on `p.opts.BufferVersion == v1` ? I see p.bufferV1 is always initialised irrespective of the opts flag